### PR TITLE
Chore: Adjust folder service methods to use cmd

### PR DESCRIPTION
**Pull request by yangkb09**

### CODE REVIEW

File: pkg/services/folder/foldertest/foldertest.go [hunk 1]
```diff
@@ -31,8 +31,7 @@ func (s *FakeService) GetFolderByTitle(ctx context.Context, user *user.SignedInU
 func (s *FakeService) CreateFolder(ctx context.Context, cmd *folder.CreateFolderCommand) (*models.Folder, error) {
 	return s.ExpectedFolder, s.ExpectedError
 }
-func (s *FakeService) UpdateFolder(ctx context.Context, user *user.SignedInUser, orgID int64, existingUid string, cmd *models.UpdateFolderCommand) error {
-	cmd.Result = s.ExpectedFolder
```
Comment (yangkb09) [inline pkg/services/folder/foldertest/foldertest.go:35]: is it ok to remove this? our `folder.UpdateFolderCommand` doesn't have Result, and we dont return anything from `UpdateFolder`either

File: pkg/services/folder/model.go [hunk 2]
```diff
@@ -56,6 +56,7 @@ type UpdateFolderCommand struct {
 	NewUID         *string `json:"uid"`
 	NewTitle       *string `json:"title"`
 	NewDescription *string `json:"description"`
+	ExistingUid    string  `json:"existinguid"`
```
Comment (yangkb09) [inline pkg/services/folder/model.go:59]: is it ok to add this? i need to pass an existing uid in pkg/services/folder/folderimpl/folder.go on ln 217 but i think i'm missing something somewhere

Comment (mildwonkey) [reply pkg/services/folder/model.go:59]: The existing folder is in the `Folder` field and you can get its UID from that.

Comment (yangkb09) [reply pkg/services/folder/model.go:59]: perfect, thanks!

File: pkg/services/folder/service.go [hunk 3]
```diff
@@ -12,8 +12,8 @@ type Service interface {
 	GetFolderByID(ctx context.Context, user *user.SignedInUser, id int64, orgID int64) (*models.Folder, error)
 	GetFolderByUID(ctx context.Context, user *user.SignedInUser, orgID int64, uid string) (*models.Folder, error)
 	GetFolderByTitle(ctx context.Context, user *user.SignedInUser, orgID int64, title string) (*models.Folder, error)
-	CreateFolder(ctx context.Context, user *user.SignedInUser, orgID int64, title, uid string) (*models.Folder, error)
-	UpdateFolder(ctx context.Context, user *user.SignedInUser, orgID int64, existingUid string, cmd *models.UpdateFolderCommand) error
+	CreateFolder(ctx context.Context, cmd *CreateFolderCommand) (*models.Folder, error)
+	UpdateFolder(ctx context.Context, cmd *UpdateFolderCommand) (*models.Folder, error)
```
Comment (yangkb09) [inline pkg/services/folder/service.go:16]: do we want to return `*models.Folder` here or `*Folder`?

File: pkg/services/dashboards/dashboard.go [hunk 4]
```diff
@@ -89,5 +88,5 @@ type FolderStore interface {
 	// GetFolderByUID retrieves a folder by its UID
 	GetFolderByUID(ctx context.Context, orgID int64, uid string) (*models.Folder, error)
 	// GetFolderByID retrieves a folder by its ID
-	GetFolderByID(ctx context.Context, orgID int64, id int64) (*folder.Folder, error)
+	GetFolderByID(ctx context.Context, orgID int64, id int64) (*models.Folder, error)
```
Comment (papagian) [inline pkg/services/dashboards/dashboard.go:91]: why we need `*models.Folder` here?

File: pkg/services/folder/service.go [hunk 5]
```diff
@@ -9,12 +9,13 @@ import (
 
 type Service interface {
 	GetFolders(ctx context.Context, user *user.SignedInUser, orgID int64, limit int64, page int64) ([]*models.Folder, error)
-	GetFolderByID(ctx context.Context, user *user.SignedInUser, id int64, orgID int64) (*models.Folder, error)
+	// GetFolderByID(ctx context.Context, user *user.SignedInUser, id int64, orgID int64) (*models.Folder, error)
+	GetFolderByID(ctx context.Context, cmd *GetFolderQuery) (*Folder, error)
```
Comment (yangkb09) [inline pkg/services/folder/service.go:13]: should i use `GetFolderQuery` as the cmd here?

Comment (mildwonkey) [reply pkg/services/folder/service.go:13]: Yes indeed!

File: pkg/api/folder.go [hunk 6]
```diff
@@ -118,12 +119,12 @@ func (hs *HTTPServer) CreateFolder(c *models.ReqContext) response.Response {
 	if err := web.Bind(c.Req, &cmd); err != nil {
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
-	folder, err := hs.folderService.CreateFolder(c.Req.Context(), c.SignedInUser, c.OrgID, cmd.Title, cmd.Uid)
+	folder, err := hs.folderService.CreateFolder(c.Req.Context(), &folder.CreateFolderCommand{Title: cmd.Title, UID: cmd.Uid})
```
Comment (yangkb09) [inline pkg/api/folder.go:122]: pass OrgID (rebase)

File: pkg/api/folder.go [hunk 7]
```diff
@@ -175,7 +177,7 @@ func (hs *HTTPServer) DeleteFolder(c *models.ReqContext) response.Response { //
 	}
 
 	uid := web.Params(c.Req)[":uid"]
-	f, err := hs.folderService.DeleteFolder(c.Req.Context(), c.SignedInUser, c.OrgID, uid, c.QueryBool("forceDeleteRules"))
+	f, err := hs.folderService.DeleteFolder(c.Req.Context(), &folder.DeleteFolderCommand{UID: uid})
```
Comment (yangkb09) [inline pkg/api/folder.go:180]: add orgID

Comment (yangkb09) [reply pkg/api/folder.go:180]: hm, `folder.DeleteFolderCommand` doesn't have an OrgID field

File: pkg/services/folder/folderimpl/folder.go [hunk 8]
```diff
@@ -207,81 +215,86 @@ func (s *Service) CreateFolder(ctx context.Context, user *user.SignedInUser, org
...
 		if err != nil {
 			return nil, toFolderError(err)
 		}
 		return nil, dashboards.ErrFolderAccessDenied
 	}
 
-	deleteCmd := models.DeleteDashboardCommand{OrgId: orgID, Id: dashFolder.Id, ForceDeleteFolderRules: forceDeleteRules}
+	folder := f.ConvertModelFolderToFolder(dashFolder)
+
+	deleteCmd := models.DeleteDashboardCommand{OrgId: user.OrgID, Id: dashFolder.Id, ForceDeleteFolderRules: forceDeleteRules} // how can i get forceDeleteRules from?
```
Comment (yangkb09) [inline pkg/services/folder/folderimpl/folder.go:291]: how can i access `forceDeleteRules` here?

File: pkg/api/folder.go (hunk 7 above)
Comment (idafurjes) [reply pkg/api/folder.go:184]: that should be added on the struct

File: pkg/services/folder/folderimpl/folder.go (hunk 8 above)
Comment (papagian) [reply pkg/services/folder/folderimpl/folder.go:291]: this flag is [for deleting the alert rules under the folder or return error if there are any](https://github.com/grafana/grafana/pull/36427/files#diff-9b8062a98126348a456e2b1a6f09d54537cf6021c6b1e370f511f7adde4a2b71).
AFAICT deleting the folder contents during folder deletion is an open discussion but I would suggest for now adding an optional boolean flag `DeleteContent` on `folder.DeleteFolderCommand`
and set it in the API level [as it happens today](https://github.com/grafana/grafana/blob/92b5baba55e42c9898a1cbf38bd083b36063d221/pkg/api/folder.go#L99).

**Pull request closed without merging**